### PR TITLE
tests: Remove dump when cluster-up fails

### PR DESCRIPTION
**Pull request by oshoval**
### What this PR does
"make dump" needs a valid cluster KUBECONFIG,
so if cluster-up failed, it doesn't make sense.
It also hides and error, and we need to scroll a bit to find the fact cluster-up failed, instead seeing it directly.

Example https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/11763/pull-kubevirt-e2e-k8s-1.29-sig-compute-migrations/1782065904025079808

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note

```

### CODE REVIEW

**Review by brianmcarey — COMMENTED** [review]
/lgtm

File: automation/test.sh [hunk 1]
```diff
@@ -274,7 +274,7 @@ export IMAGE_PREFIX_ALT=${IMAGE_PREFIX_ALT:-kv-}
 
 build_images
 
-trap '{ collect_debug_logs; echo "Dump kubevirt state:"; make dump; }' ERR
+trap '{ collect_debug_logs; }' ERR
```
Comment (xpivarc) [inline automation/test.sh:277]: Don't we want to move it below?

Comment (brianmcarey) [reply automation/test.sh:277]: There are `make dump`s further below in the script.

Comment (brianmcarey) [reply automation/test.sh:277]: `make cluster-sync` has its own error handling - https://github.com/kubevirt/kubevirt/blob/ee0144cfd43170ac12bf973867c93850f78697d5/hack/cluster-deploy.sh#L72

**Pull request merged**